config/locales/en.yml

```diff
@@ -0,0 +1,33 @@
+# Files in the config/locales directory are used for internationalization
```

Comment:
Whats this file about?

Reply:
Defines the key and translation for the en locale.

For this example, the configured locale is `en` and the setting is on key `hello`
```
en:
    hello: "Hello world!"
```
To use this in the view, we use the format `<%= t('hello') %>` and it will display `Hello world!` on the page.

Reply:
Can you elaborate more on how translations work in rails?

Reply:
Translations in Rails work by:
- Translation strings for various languages can be stored as key value pairs in different `.yml` files, e.g. `en.yml`, `es.yml`, `pt.yml`, etc.
- Setting a default locale to use, and what it will fallback to, for instance `en`
- When a different language is selected, this can be either done via params in the url, e.g. `?lang=es`, HTTP headers, routes `https://example.com/es`, domains `https://es.example.com` etc., `i18n` will replace all the `t(' ')` keys with their respective values based on the correct `.yml` file
- If the respective keys can't be found, it can fallback to a default locale's key

Reply:
> When a different language is selected, this can be either done via params in the url
What if I don't want to pass it like that. Say, I have a service class that wants to return a translated value. How can I do it?

Reply:
Do you mean like still being on the same URL, without changing the global locale setting but returning a translated value depending on certain conditions?

Reply:
I can run `I18n.locale = 'es'` followed by `helper.t('my-string')` to return the translated value for 'es'.